src/search.cpp

```diff
@@ -387,7 +391,51 @@ void MainThread::search() {
       for (Thread* th : Threads)
           if (   th->completedDepth > bestThread->completedDepth
               && th->rootMoves[0].score > bestThread->rootMoves[0].score)
-              bestThread = th;
+              longestPVThread = bestThread = th;
+
+      if (bestThread->rootMoves[0].pv.size() < minPlies) {
+          for (Thread* th : Threads) {
+              // Look for the best thread that meets the minimum move criteria.
+              // Don't take a thread unless it's within the appropriate range of score
+              // eval. 
+              if (th->rootMoves[0].pv.size() <= bestThread->rootMoves[0].pv.size()) {
+                  continue;
+              }
+              bool possiblePVReplacement = true;
```

Comment:
It might be simpler to read and easier for the compiler to optimize using `std::mismatch` http://www.cplusplus.com/reference/algorithm/mismatch/